### PR TITLE
Fix mongoid last sort

### PR DESCRIPTION
**Pull request by jakolehm**
Quarantees order when last item is fetched from a collection.

### CODE REVIEW

**Comment by SpComb** [comment]
Is it possible to have a rspec for these, or it is hard to repro?

**Comment by jakolehm** [comment]
I'm surprised that our tests did not catch this. It might be a bit hard to reproduce this in tests but I can try.

**Comment by jakolehm** [comment]
Refactored & did some cleanup because linter was showing quite many red lines. @jnummelin @SpComb PTAL.

File: server/spec/models/host_node_stat_spec.rb [hunk 1]
```diff
@@ -11,6 +11,21 @@
...
   it { should have_index_for(host_node_id:1, created_at: 1).with_options(background: true) }
 
+  describe '.latest' do
+    let(:grid) { Grid.create!(name: 'test-grid') }
+    let(:node) { HostNode.create!(grid: grid, name: 'test-node') }
+
+    it 'returns latest stat item' do
+      node.host_node_stats.create
+      last = node.host_node_stats.create
+      expect(described_class.latest).to eq(last)
```
Comment (SpComb) [inline server/spec/models/host_node_stat_spec.rb]: Confirm that the spec works, aka even with just the two items, removing the `asc(:id)` allows this to fail:

```
  1) HostNodeStat.latest returns latest stat item
     Failure/Error: expect(described_class.latest).to eq(last)
       
       expected: #<HostNodeStat _id: 5955f959de3578646bd0b488, created_at: 2017-06-30 07:10:17 UTC, load: nil, memory: nil, filesystem: nil, usage: nil, cpu: nil, network: nil, grid_id: nil, host_node_id: BSON::ObjectId('5955f959de3578646bd0b484')>
            got: #<HostNodeStat _id: 5955f959de3578646bd0b487, created_at: 2017-06-30 07:10:17 UTC, load: nil, memory: nil, filesystem: nil, usage: nil, cpu: nil, network: nil, grid_id: nil, host_node_id: BSON::ObjectId('5955f959de3578646bd0b484')>
       
       (compared using ==)
       
       Diff:
       @@ -1,2 +1,2 @@
       -#<HostNodeStat _id: 5955f959de3578646bd0b488, created_at: 2017-06-30 07:10:17 UTC, load: nil, memory: nil, filesystem: nil, usage: nil, cpu: nil, network: nil, grid_id: nil, host_node_id: BSON::ObjectId('5955f959de3578646bd0b484')>
       +#<HostNodeStat _id: 5955f959de3578646bd0b487, created_at: 2017-06-30 07:10:17 UTC, load: nil, memory: nil, filesystem: nil, usage: nil, cpu: nil, network: nil, grid_id: nil, host_node_id: BSON::ObjectId('5955f959de3578646bd0b484')>
       
     # ./spec/models/host_node_stat_spec.rb:21:in `block (3 levels) in <top (required)>'
```

**Comment by SpComb** [comment]
This is probably something that broke with the  #2257 mongoid/mongo-driver upgrade in 1.3.0?

> first and last no longer add an _id sort when no sorting options have been provided. In order to guarantee that a document is the first or last, it needs to now contain an explicit sort.

https://docs.mongodb.com/mongoid/5.2/tutorials/mongoid-upgrade/

**Pull request merged**